kola/tests/ostree/basic.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ostree
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/kola/tests/rpmostree"
+	"github.com/coreos/mantle/platform"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         ostreeBasicTest,
+		ClusterSize: 1,
+		Name:        "rhcos.ostree.basic",
+		Distros:     []string{"rhcos", "fcos"},
+		FailFast:    true,
+	})
+
+	register.Register(&register.Test{
+		Run:         ostreeRemoteTest,
+		ClusterSize: 1,
+		Name:        "rhcos.ostree.remote",
+		Distros:     []string{"rhcos", "fcos"},
+		ExcludePlatforms: []string{"qemu"}, // need network to contact remote
+		FailFast:    true,
+	})
+}
+
+type ostreeAdminStatus struct {
+	Checksum string
+	Version  string
+	Origin   string
+}
+
+// getOstreeAdminStatus stuffs the important output of `ostree admin status`
+// into an `ostreeAdminStatus` struct
+func getOstreeAdminStatus(c cluster.TestCluster, m platform.Machine) (ostreeAdminStatus, error) {
+	oaStatus := ostreeAdminStatus{}
+
+	oasOutput, err:= c.SSH(m, "ostree admin status")
+	if err != nil {
+		return oaStatus, fmt.Errorf(`Could not get "ostree admin status": %v`, err)
+	}
+
+	oasSplit := strings.Split(string(oasOutput), "\n")
+	if len(oasSplit) != 3 {
+		return oaStatus, fmt.Errorf(`Unexpected output from "ostree admin status": %v`, string(oasOutput))
+	}
+
+	// take the first line of `ostree admin status`, split on whitespace,
+	// return the csum+serial, trim the serial suffix to get the csum
+	oaStatus.Checksum = strings.TrimSuffix(strings.Fields(oasSplit[0])[2], ".0")
```

Comment:
I'm a bit weary about adding direct index references without size checks. Could we either a) split this out and add checks or b) switch to regex where a `nil` match is checked?

Same with the `Version` & `Origin` assignments below.